src/dbnode/storage/namespace.go

```diff
@@ -1231,12 +1232,18 @@ func (n *dbNamespace) WarmFlush(
 			detailedErr := fmt.Errorf("shard %d failed to flush data: %v",
 				shard.ID(), err)
 			multiErr = multiErr.Add(detailedErr)
+
+			// Errors so mark as failed here.
+			// We mark as success later in time to ensure that happens after index flushing as well.
+			shard.MarkWarmFlushStateSuccessOrError(blockStart, err)
+		} else {
+			flushedShards = append(flushedShards, shard)
 		}
```

Comment:
As per other suggestions on the PR - use an early return here as per Effective Go?
```suggestion
			// Errors so mark as failed here.
			// We mark as success later in time to ensure that happens after index flushing as well.
			shard.MarkWarmFlushStateSuccessOrError(blockStart, err)
			continue
		}
		
		flushedShards = append(flushedShards, shard)
```